cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -532,12 +533,15 @@ private static void prepareMirrorMaker2ConnectorConfig(Reconciliation reconcilia
     private static String oauthJaasConfig(KafkaMirrorMaker2ClusterSpec cluster,
                                           Map<String, String> authProperties,
                                           KafkaClientAuthenticationOAuth oauth) {
-        Map<String, String> jaasOptions = new HashMap<>();
         String oauthConfig = authProperties.get(AuthenticationUtils.OAUTH_CONFIG);
-        var index = oauthConfig.indexOf("=");
-        if (index != -1) {
-            jaasOptions.put(oauthConfig.substring(0, index), oauthConfig.substring(index + 1));
-        }
+        // convert the sasl.jaas.config string value to a map of jaas options
+        Map<String, String> jaasOptions = Arrays.stream(
+            oauthConfig.replace("\"", "").split(" "))
+                .map(option -> option.split("="))
+                .filter(entry -> entry.length == 2)
+                .collect(Collectors.toMap(entry -> entry[0], entry -> entry[1])
+        );
```

Comment:
This _works_, but it's really ugly. Before this method is called, `authProperties` gets created with the `OAUTH_CONFIG` key added with a carefully constructed JAAS string, and here we end up parsing it back to keys and values. I was looking at whether we should refactor to avoid formatting+parsing, and pass around something more structured than a `String`. It ends up being a bigger refactoring, but leaving the code slightly more clear and direct.

Reply:
Yes, I agree. It can be improved. I'll leave it to @ShazaAldawamneh.

Reply:
working on it.

Reply:
thanks for your feedback Tom and Fede